jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -282,7 +281,7 @@ ecma_builtin_number_prototype_object_to_string (ecma_number_t this_arg_number, /
     return ecma_raise_range_error (ECMA_ERR_MSG ("Radix must be between 2 and 36."));
   }
 
-  if (radix == 10)
+  if (radix == 10 || ecma_number_is_zero (this_arg_number))
```

Comment:
IMO this check is not correct. What about other radix values? The source of the problem comes from `if (radix < 2 || radix > 36)` check does not handle `NaN` after the toNumber conversion. That's why toInteger should be used here.